examples/CMakeLists.txt

```diff
@@ -11,6 +11,10 @@ if(protobuf_VERBOSE)
   message(STATUS "Using Protocol Buffers ${Protobuf_VERSION}")
 endif()
 
+if(protobuf_WITH_ZLIB)
+  find_package(ZLIB)

```

Comment:
I considered that, but I think what we should actually do is remove ZLIB
from the interface dependencies. I believe users only need to link with
ZLIB if they use the gzipoutput class

On Thu, Jun 9, 2016, 8:07 PM Konstantin Podsvirov <notifications@github.com>
wrote:

> In examples/CMakeLists.txt
> <https://github.com/google/protobuf/pull/1667#discussion_r66556631>:
>
> > @@ -11,6 +11,10 @@ if(protobuf_VERBOSE)
> >    message(STATUS "Using Protocol Buffers ${Protobuf_VERSION}")
> >  endif()
> >
> > +if(protobuf_WITH_ZLIB)
> > +  find_package(ZLIB)
>
> This is a common issue associated with imported targets and finding
> dependencies, which I still do not fully understand.
> I have suggested that we can put this code in protobuf-config.cmake and
> relieve the user of this work is. But I do not know the true test of it or
> not?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/google/protobuf/pull/1667/files/7ff2217a7e4c3f3a18878b39b37194c0090bc286#r66556631>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAVm3AIvJC_WJoDv6OI6qcFk0kWJFgl6ks5qKNULgaJpZM4Iyh9->
> .
>